src/Symfony/Component/Form/Extension/Core/Type/RepeatedType.php

```diff
@@ -36,8 +37,8 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 $options['first_name'],
                 $options['second_name'],
             ]))
-            ->add($options['first_name'], $options['type'], array_merge($options['options'], $options['first_options']))
-            ->add($options['second_name'], $options['type'], array_merge($options['options'], $options['second_options']))
+            ->add($options['first_name'], $options['type'], $this->mergeOptions($options['options'], $options['first_options']))
```

Comment:
Instead of adding a private method I would add above:
```php
// children fields must always be mapped
$defaultOptions = ['mapped' => false];
```
Then do the following:
```suggestion
            ->add($options['first_name'], $options['type'], array_merge($options['options'], $options['first_options'], $defaultOptions))
```

Reply:
Well, with `true` rather than `false` of course :smile:

Reply:
Edited, thanks x)

Reply:
Keep in mind that removing this will hide the behavior when developer specify `mapped => false` to one of the inner fields.

Reply:
Yes but we need a consistent fix. Either throw in all case or override in all case.
I say this PR deserves a doc update, adding a note in the `RepeatedType` reference to tell about that overriding as a requirement for this type to work properly. WDYT? Are you up to it too :) or should we take care of this one?

Reply:
Also note that it would be a lot more complex to try to throw in all cases.

Reply:
I'm fine with silently overriding in all cases here (we might add a note in the doc saying the inner fields cannot be unmapped, and that you should use `mapped => false` on the RepeatedType itself if needed).

Reply:
done